services/management/ia/channels_service.go

```diff
@@ -103,156 +103,164 @@ func (s *ChannelsService) ListChannels(ctx context.Context, request *iav1beta1.L
+			Channel:      req.SlackConfig.Channel,
 		}
 	}
-	if webhookConf := req.GetWebhookConfig(); webhookConf != nil {
+	if req.WebhookConfig != nil {
 		params.WebHookConfig = &models.WebHookConfig{
-			SendResolved: webhookConf.SendResolved,
-			URL:          webhookConf.Url,
-			MaxAlerts:    webhookConf.MaxAlerts,
-			HTTPConfig:   convertHTTPConfigToModel(webhookConf.HttpConfig),
+			SendResolved: req.WebhookConfig.SendResolved,
+			URL:          req.WebhookConfig.Url,
+			MaxAlerts:    req.WebhookConfig.MaxAlerts,
+			HTTPConfig:   convertHTTPConfigToModel(req.WebhookConfig.HttpConfig),
 		}
 	}
 
-	channel, err := s.ia.AddChannel(params)
-	if err != nil {
-		return nil, err
+	var channel *models.Channel
+	e := s.db.InTransaction(func(tx *reform.TX) error {
+		var err error
+		channel, err = models.CreateChannel(tx.Querier, params)
+		return err
+	})
+	if e != nil {
+		return nil, e
 	}
-
 	return &iav1beta1.AddChannelResponse{ChannelId: channel.ID}, nil
 }
 
 // ChangeChannel changes existing notification channel.
 func (s *ChannelsService) ChangeChannel(ctx context.Context, req *iav1beta1.ChangeChannelRequest) (*iav1beta1.ChangeChannelResponse, error) {
 	params := &models.ChangeChannelParams{
-		Disabled: req.GetDisabled(),
+		Disabled: req.Disabled,
 	}
 
-	if emailConf := req.GetEmailConfig(); emailConf != nil {
+	if c := req.EmailConfig; c != nil {
 		params.EmailConfig = &models.EmailConfig{
-			SendResolved: emailConf.SendResolved,
-			To:           emailConf.To,
+			SendResolved: c.SendResolved,
+			To:           c.To,
 		}
 	}
-	if pagerDutyConf := req.GetPagerdutyConfig(); pagerDutyConf != nil {
+	if c := req.PagerdutyConfig; c != nil {
 		params.PagerDutyConfig = &models.PagerDutyConfig{
-			SendResolved: pagerDutyConf.SendResolved,
-			RoutingKey:   pagerDutyConf.RoutingKey,
-			ServiceKey:   pagerDutyConf.ServiceKey,
+			SendResolved: c.SendResolved,
+			RoutingKey:   c.RoutingKey,
+			ServiceKey:   c.ServiceKey,
 		}
 	}
-	if slackConf := req.GetSlackConfig(); slackConf != nil {
+	if c := req.SlackConfig; c != nil {
 		params.SlackConfig = &models.SlackConfig{
-			SendResolved: slackConf.SendResolved,
-			Channel:      slackConf.Channel,
+			SendResolved: c.SendResolved,
+			Channel:      c.Channel,
 		}
 	}
-	if webhookConf := req.GetWebhookConfig(); webhookConf != nil {
+	if c := req.WebhookConfig; c != nil {
 		params.WebHookConfig = &models.WebHookConfig{
-			SendResolved: webhookConf.SendResolved,
-			URL:          webhookConf.Url,
-			MaxAlerts:    webhookConf.MaxAlerts,
-			HTTPConfig:   convertHTTPConfigToModel(webhookConf.HttpConfig),
+			SendResolved: c.SendResolved,
+			URL:          c.Url,
+			MaxAlerts:    c.MaxAlerts,
+			HTTPConfig:   convertHTTPConfigToModel(c.HttpConfig),
 		}
 	}
 
-	err := s.ia.ChangeChannel(req.ChannelId, params)
-	if err != nil {
-		return nil, err
+	e := s.db.InTransaction(func(tx *reform.TX) error {
+		_, err := models.ChangeChannel(tx.Querier, req.ChannelId, params)
+		return err
+	})
```

Comment:
I have a question not regarding that PR, but in general. `ChangeChannel` accepts `reform.Querier`, however, it can be transactional and non-transactional. So function interface doesn't descriptive enough, is there any way to explicitly show that function expected to be run inside a transaction? Probably it's more about `reform` lib, maybe something like `reform.TxQuerier`.

Reply:
The idea is that all helpers should use querier, and all API handlers should use transactions. We often skip that when we know that there is only query, but, I think we could add "all API handlers should use transaction" as a general rule

Reply:
cf93767acee8bdda7281d9a583ef515fe7089dbf

Reply:
This is still an implicit contract, but ok. Thanks)